src/openzaak/components/zaken/models/utils.py

```diff
@@ -162,21 +140,16 @@ def get_brondatum(zaak: Zaak, afleidingswijze: str, datum_kenmerk: str=None,
         raise NotImplementedError
 
     elif afleidingswijze == BrondatumArchiefprocedureAfleidingswijze.ingangsdatum_besluit:
-        zaakbesluiten = zaak.zaakbesluit_set.all()
+        zaakbesluiten = zaak.besluit_set.all()
         if not zaakbesluiten.exists():
             # Cannot use ingangsdatum_besluit if Zaak has no Besluiten
             raise DetermineProcessEndDateException(
                 _('Geen besluiten aan zaak gekoppeld om brondatum uit af te leiden.')
             )
 
-        Client = import_string(settings.ZDS_CLIENT_CLASS)
-        client = Client.from_url(zaakbesluiten.first().besluit)
-        client.auth = APICredential.get_auth(zaakbesluiten.first().besluit)
-
         max_ingangsdatum = None
-        for zaakbesluit in zaakbesluiten:
-            related_besluit = client.retrieve('besluit', url=zaakbesluit.besluit)
-            ingangsdatum = datetime.strptime(related_besluit['ingangsdatum'], '%Y-%m-%d')
+        for besluit in zaakbesluiten:
```

Comment:
should be replaced with a `Max("ingangsdatum")` aggregation query I think.

Reply:
aggregation is added to the query